src/vivarium_nih_us_cvd/data/loader.py

```diff
@@ -370,74 +381,276 @@ def _get_ihd_sequela() -> Dict[str, List["Sequela"]]:
+        "location_name==@location & sim_cause==@heart_failure_type"
+    )
+
+    hf_proportions = hf_proportions[PROPORTION_DATA_INDEX_COLUMNS + ["proportion"]]
+
+    return hf_proportions
+
+
+def get_proportion_adjusted_heart_failure_data(
+    location: str, heart_failure_type: str, measure: str
+) -> pd.DataFrame:
+    # pull measure data
+    location_id = utility_data.get_location_id(location)
+    heart_failure_data = gbd.get_modelable_entity_draws(
+        data_values.HEART_FAILURE_ME_ID, location_id
+    )
+    measure_data = heart_failure_data[
+        heart_failure_data.measure_id == vi_globals.MEASURES[measure]
+    ]
+    measure_data = vi_utils.normalize(measure_data, fill_value=0)
+    measure_data = measure_data.filter(
+        vi_globals.DEMOGRAPHIC_COLUMNS + vi_globals.DRAW_COLUMNS
+    )
+
+    # pull proportions data
+    hf_proportions = get_heart_failure_proportions(location, heart_failure_type)
+
+    # apply proportion data
+    draw_cols = [f"draw_{i}" for i in range(1000)]
+
+    measure_data = measure_data.merge(hf_proportions, on=PROPORTION_DATA_INDEX_COLUMNS)
+    measure_data[draw_cols] = measure_data[draw_cols].mul(measure_data["proportion"], axis=0)
+
+    # format for vivarium
+    prop_adjusted_data = measure_data.filter(
+        vi_globals.DEMOGRAPHIC_COLUMNS + vi_globals.DRAW_COLUMNS
+    )
+    prop_adjusted_data = vi_utils.reshape(prop_adjusted_data)
+    prop_adjusted_data = vi_utils.scrub_gbd_conventions(prop_adjusted_data, location)
+    prop_adjusted_data = vi_utils.split_interval(
+        prop_adjusted_data, interval_column="age", split_column_prefix="age"
+    )
+    prop_adjusted_data = vi_utils.split_interval(
+        prop_adjusted_data, interval_column="year", split_column_prefix="year"
+    )
+    prop_adjusted_data = vi_utils.sort_hierarchical_data(prop_adjusted_data).droplevel(
+        "location"
+    )
+
+    return prop_adjusted_data
+
+
+def load_prevalence_mi(key: str, location: str) -> pd.DataFrame:
     ihd_seq = _get_ihd_sequela()
     map = {
-        data_keys.MYOCARDIAL_INFARCTION.PREVALENCE_ACUTE: ihd_seq["acute_mi"],
-        data_keys.MYOCARDIAL_INFARCTION.PREVALENCE_POST: ihd_seq["post_mi"],
+        data_keys.IHD_AND_HF.PREVALENCE_ACUTE_MI: (ihd_seq["acute_mi"], False),
+        data_keys.IHD_AND_HF.PREVALENCE_ACUTE_MI_AND_HF: (ihd_seq["acute_mi"], True),
+        data_keys.IHD_AND_HF.PREVALENCE_POST_MI: (ihd_seq["post_mi"], False),
     }
-    prevalence = _load_and_sum_prevalence_from_sequelae(key, map, location)
+    sequelae, prevalence_includes_heart_failure = map[key]
+    sequelae_prevalence = _load_and_sum_prevalence_from_sequelae(sequelae, location)
+
+    hf_ihd_prevalence = load_prevalence_heart_failure(
+        data_keys.IHD_AND_HF.PREVALENCE_HF_IHD, location
+    )
+
+    if prevalence_includes_heart_failure:
+        prevalence = hf_ihd_prevalence * sequelae_prevalence
+    else:
+        prevalence = (1 - hf_ihd_prevalence) * sequelae_prevalence
```

Comment:
So the sum of these three prevalences is the prevalence of the full envelope?

Reply:
Yeah I talked to Syl that this what we wanted since I was also concerned about this. It's a bit tricky because if you experience acute myocardial infarction AND heart failure, your "post MI" state is actually heart failure from ischemic heart disease.